crates/sui-core/src/consensus_validator.rs

```diff
@@ -141,6 +115,55 @@ impl TransactionValidator for SuiTxValidator {
     }
 }
 
+fn tx_from_bytes(tx: &[u8]) -> Result<ConsensusTransaction, eyre::Report> {
+    bcs::from_bytes::<ConsensusTransaction>(tx)
+        .wrap_err("Malformed transaction (failed to deserialize)")
+}
+
+#[async_trait]
+impl TransactionValidator for SuiTxValidator {
+    type Error = eyre::Report;
+
+    fn validate(&self, _tx: &[u8]) -> Result<(), Self::Error> {
+        // We only accept transactions from local sui instance so no need to re-verify it
+        Ok(())
+    }
+
+    async fn validate_batch(
+        &self,
+        b: &narwhal_types::Batch,
+        protocol_config: &ProtocolConfig,
+    ) -> Result<(), Self::Error> {
+        let _scope = monitored_scope("ValidateBatch");
+
+        // TODO: Remove once we have removed BatchV1 from the codebase.
+        validate_batch_version(b, protocol_config)
+            .map_err(|err| eyre::eyre!(format!("Invalid Batch: {err}")))?;
+
+        let txs = b
+            .transactions()
+            .iter()
+            .map(|tx| tx_from_bytes(tx).map(|tx| tx.kind))
+            .collect::<Result<Vec<_>, _>>()?;
+
+        self.validate_transactions(txs).await
+    }
+}
+
+#[async_trait]
+impl BlockValidator for SuiTxValidator {
```

Comment:
Nice, this aligns well with the expectations of https://github.com/MystenLabs/sui/pull/14559